### DISCUSSION TITLE
To Woosh or not to Woosh

### DISCUSSION THREAD

**Discussion by cclauss** [django-haystack/django-haystack#1961] [Ideas]
As discussed in:
* #1924
* #1929 
* #1943 
* #1945 

Our dependency [`Whoosh`](https://pypi.org/project/Whoosh) is no longer maintained and has not been upgraded on PyPI since 2016.
There is a new alternative [`Whoosh-Reloaded`](https://pypi.org/project/Whoosh-Reloaded) but it has created new failures in our unittests.

My project does not use `Whoosh` so I have been trying to remove that dependency but others tell me that their projects still use Whoosh.  My short-term advice is to try to get our tests to pass with the current version of `Whoosh` with as minimal changes as possible.

Once that is done, we can consider if we should investigate if we should maintain our own fork of Whoosh or if we should adopt `Whoosh-Reloaded`.

@black3r @claudep @acdha

**Comment by claudep** [comment]
@ZeroCool940711, could you maybe elaborate about your commitment on maintaining Whoosh-Reloaded in the longer term? For now, looking at the commit log shows some real activities on about 4 days, so you may understand that from an outside point of view, that looks fragile.

**Reply by cclauss** [reply]
#1962 is a simple fix that enables our Woosh-related tests to pass again on GitHub Actions.
* #1962

**Comment by claudep** [comment]
Yes, but sooner or later, we will be confronted with a version incompatibility...

**Reply by cclauss** [reply]
# 💯
